.github/workflows/python-tests.yml

```diff
@@ -0,0 +1,28 @@
+name: Run Python Tests
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - src/config/**
+      - src/**.py
+      - src/requirements.txt
+jobs:
+  tests:
+    name: Run pytest
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+      - name: Install Requirements
+        run: |
+          python -m pip install --upgrade pip
```

Comment:
Do we really need to upgrade pip here? If the system can work with what `setup-python` ships with, then we can avoid yet another process that involves network requests.

Reply:
I was taking it from their documentation: https://docs.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions
It's fairly quick so happy to leave in.